packages/lib/src/FormController/FormController.ts

```diff
@@ -413,81 +410,89 @@ export class FormController {
+    const errors = (await (async () => {
       try {
         const result = await this.runFieldLevelValidation(fieldName);
         if (result !== undefined && result !== null) {
           return result;
         }
+        return undefined;
       } catch (e) {
         return e;
       }
-    })();
+    })()) as FieldErrors;
 
-    runInAction(() => (fieldMeta.isValidating = false));
-    this.setFormValidationErrors(merge(this.formValidationErrors, {[fieldName]: errors}));
-    this.updateFieldErrors(field, errors);
+    runInAction(() => {
+      const field = this.fields.get(fieldName)!;
+      this.setFieldMeta(field, {
+        isValidating: false,
+      });
+      this.setFormValidationErrors(merge(this.API.errors, {[fieldName]: errors}));
+      if (field.meta.isMounted) {
+        this.setFieldErrors(field, errors);
+      }
 
-    this.setIsValidating(false);
+      this.setIsValidating(false);
+    });
   };
 
   // validates the form, by calling form level onValidate function combined with field level validations,
   // passed to Field as `validate` prop
-  validate = async () => {
+  protected validate = async () => {
     this.setIsValidating(true);
 
     const [fieldValidationErrors, formValidationErrors] = await Promise.all([
       this.runFieldLevelValidations(),
       this.runFormLevelValidations(),
     ]);
 
-    this.setFormValidationErrors(merge(fieldValidationErrors, formValidationErrors));
-
-    this.updateErrorsForEveryField(this.formValidationErrors);
-
-    this.setIsValidating(false);
+    runInAction(() => {
+      this.setFormValidationErrors(merge(fieldValidationErrors, formValidationErrors));
+      this.updateErrorsForEveryField(this.API.errors);
+      this.setIsValidating(false);
+    });
   };
 
   // wraps submit function passed as Form `onSubmit` prop after it's being passed to child render function
-  submit = async <E>(submitEvent?: React.FormEvent<E>) => {
+  @action protected submit = async <E extends HTMLElement>(submitEvent?: React.FormEvent<E>) => {
     if (submitEvent) {
       submitEvent.persist();
       submitEvent.preventDefault();
     }
 
     runInAction(() => {
-      this.setSubmitCount(this.submitCount + 1);
+      this.setSubmitCount(this.API.meta.submitCount + 1);
       this.setIsSubmitting(true);
     });
 
     await this.validate();
 
-    const errors = toJSCompat(this.formValidationErrors);
-    const values = toJSCompat(this.formattedValues);
+    const errors = toJSCompat(this.API.errors);
+    const values = toJSCompat(this.API.values);
 
     try {
       if (this.options.onSubmit) {
         await this.options.onSubmit(errors, values, submitEvent);
       }
 
-      if (this.formValidationErrors === null) {
-        this.setInitialValuesToCurrentValues();
+      if (isEmpty(this.API.errors)) {
+        runInAction(() => {
+          this.setInitialValuesToCurrentValues();
+          this.setIsSubmitting(false);
+        });
       }
-    } finally {
+    } catch {
```

Comment:
change to finally, `isSubmitting` stay true in case of this.API.errors not empty